### PR TITLE
Feature/99  list

### PR DESCRIPTION
**Pull request by joesnellpdx**
* Kitchen sink file showing all variations in all states, including too much and too little content
* Macro to create all variations
* Update tokens file with necessary style decisions

### CODE REVIEW

**Comment by joesnellpdx** [comment]
@kevinmpowell @jamesmelzer @patrickmckowen 

Initial PR ready for lists.
Expecting iteration.

Created two sink pages:

1. Main sink page: `/sink-pages/components/lists.html`

2. Grid examples - pushing the list past its limits: `sink-pages/components/list-grids.html`

**Comment by joesnellpdx** [comment]
@kevinmpowell  @patrickmckowen @jamesmelzer 

Sizes added. Ready for review.

See:
`/sink-pages/components/lists.html`
`/sink-pages/components/list-sizes.html`
`/sink-pages/components/list-grids.html` 

Questions / Notes:
- List includes **Media Object** https://github.com/JDRF/spirit/issues/107
- Check macro construction, variables, ouput
- Check list variables - new tokens needed?
- Breakpoint used for sizes matches heading breakpoints ( i.e. `spririt-h4`) - 1024px
- Default is set to Extra Large  as this is follows the `spririt-h4` styles from small to large breakpoints
- Should default become 'medium' and large to small and medium to extra small? Semantics does not feel correct as is ( or should styles change / other update )?
- Is the list grids page redundant with some grid options added to the sizes page? Please review across breakpoints to see where styles 'break.'

File: library/components/list/list.njk [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  title_tag='h3',
+  link=false,
+  link_text="View",
+  link_secondary=false,
+  link_secondary_text="Donate",
+  image_src=false,
+  image_right=false,
+  avatar_src=false,
+  icon_name=false,
+  avatar_src=false,
```
Comment (jamesmelzer) [inline library/components/list/list.njk]: - [ ] I see avatar_src twice.

File: library/components/list/list.scss [hunk 2]
```diff
@@ -0,0 +1,263 @@
+// List Variables
+$spirit-list-text-light: $spirit-color-grey-35;
+$spirit-list-border-color: $spirit-color-grey-95;
+$spirit-list-border-width: 1px;
+$spirit-image-width: 72px;
+$sprirt-image-width-l: 180px;
+$sprirt-image-height-l: 120px;
+$spirit-list-avatar-width: $spirit-space-generic-6-x;
+$spirit-list-icon-size: 20px;
```
Comment (jamesmelzer) [inline library/components/list/list.scss]: - [ ] $spirit-size-icon-s (20px)

File: library/components/list/list.scss [hunk 3]
```diff
@@ -0,0 +1,263 @@
+// List Variables
+$spirit-list-text-light: $spirit-color-grey-35;
+$spirit-list-border-color: $spirit-color-grey-95;
+$spirit-list-border-width: 1px;
+$spirit-image-width: 72px;
+$sprirt-image-width-l: 180px;
```
Comment (jamesmelzer) [inline library/components/list/list.scss]: - [ ] "sprirt" => "spirit" lines 6 and 7

File: library/components/list/list.scss [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+.spirit-list-item__image-contain {
+  background-color: $spirit-brand-color-primary;
+  border-radius: $spirit-border-radius-l;
+  flex: 0 0 $spirit-image-width;
+  height: $spirit-image-width;
+  margin-right: $spirit-space-generic-2-x;
+  overflow: hidden;
+  width: $spirit-image-width;
+  @media screen and (min-width: $spirit-list-breakpoint) {
+    flex-basis: $sprirt-image-width-l;
```
Comment (jamesmelzer) [inline library/components/list/list.scss]: - [ ] sprirt = spirit a few times in here too...

File: library/components/list/list.scss [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+  flex-flow: row nowrap;
+  padding: $spirit-space-inset-2-x;
+  position: relative;
+
+  &:after {
+    background: $spirit-list-border-color;
+    bottom: 0;
+    content: '';
+    height: $spirit-list-border-width;
+    left: $spirit-space-generic-2-x;
```
[the remark this answers is not in the document]
Comment (joesnellpdx) [reply library/components/list/list.scss]: @jamesmelzer This has been updated. Only bottom border affected.

**Comment by joesnellpdx** [comment]
@jamesmelzer Updates per your comments complete and pushed to branch.

File: library/components/list/list.njk [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  </{{tag}}>
+{% endmacro %}
+
+{# List Item #}
+{% macro list_item(
+  class=false,
+  title_text="List Item",
+  title_meta=false,
+  title_tag='h3',
+  link=false,
```
Comment (kevinmpowell) [inline library/components/list/list.njk]: * [ ] Change `link` to `href` to align with previously build `vertical_navigation_list` and `card`

File: library/components/list/list.njk [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  link_text="View",
+  link_secondary=false,
```
Comment (kevinmpowell) [inline library/components/list/list.njk]: * [ ] Change to `secondary_href` or `link_secondary_href` if you want to keep the `link_secondary` naming convention.

File: library/components/list/list.scss [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+  flex: 0 0 $spirit-list-avatar-width;
+  flex-flow: column nowrap;
+  font-size: $spirit-list-icon-size;
+  height: $spirit-list-avatar-width;
+  justify-content: center;
+  line-height: $spirit-list-icon-size;
+  margin-right: $spirit-space-generic-2-x;
+  overflow: hidden;
+  width: $spirit-list-avatar-width;
+  // &:after {
```
Comment (kevinmpowell) [inline library/components/list/list.scss]: * [ ] Remove commented out code

**Comment by joesnellpdx** [comment]
@jamesmelzer @kevinmpowell 
Code updates made.

Before merging, I want to get approval from @patrickmckowen.

I know @patrickmckowen and @ismaeltoure7 wanted to view spacing in browser for lists. Need confirmation we are okay or if there are changes needed.

**Comment by joesnellpdx** [comment]
Adding one tweak to two sink pages. Will update.

**Comment by joesnellpdx** [comment]
Ready for review. All updated.

**Pull request merged**